src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -93,20 +93,14 @@ public CuratorFramework get() {
 
   public void start() {
     if (started.compareAndSet(false, true)) {
-      executor.submit(new Runnable() {
-
-        //@Override Java 5 compatibility
-        public void run() {
-          try {
-            fetch(true);
-          } finally {
-            executor.schedule(this, 10, TimeUnit.MINUTES);
```

Comment:
I think this `executor.schedule(this, 10, TimeUnit.MINUTES)` statement was to give us an extra chance to reconnect if something went wrong. In practice, 10 minutes could be too long to wait, and properly listening to events and refetching seems a more robust approach.

Reply:
I believe the intent was to do a background fetch every 10 minutes. This serves a few purposes:
- if we get disconnected from ZK due to an outage, this should continue attempting to reconnect us
- if our watcher somehow silently dies, this will fetch the value in the background every 10 minutes. if we discover a new version using this method, then our watcher must not be firing and we replace the Curator client